ui/hooks/useName.ts

```diff
@@ -33,11 +37,20 @@ export function useName(
   const normalizedValue = normalizeValue(value, type);
   const typeVariationKey = getVariationKey(type, chainId);
   const variationKey = variation ?? typeVariationKey;
-  const nameEntry = names[type]?.[normalizedValue]?.[variationKey];
-
+  const variationsToNameEntries = names[type]?.[normalizedValue] ?? {};
+  const nameEntry =
+    variationsToNameEntries[variationKey] ??
+    variationsToNameEntries[FALLBACK_VARIATION];
```

Comment:
@matthewwalsh0 suppose `variationsToNameEntries[variationKey]` is:
- `name===null` 
- has one or more `proposedNames`

Should we:
1. return the nameEntry as-is? OR
2. check the fallbackEntry for a name and merge it into the returned nameEntry?

Reply:
Done.